fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1540,11 +1540,14 @@ def _print_r0(self, msg: str, restart: bool = False) -> None:
 
 def _get_default_cuda_device(module: nn.Module) -> torch.device:
     """Try to infer CUDA device from module parameters."""
-    compute_device = next(module.parameters()).device
-    if compute_device.type != "cuda":
-        # Fall back to current CUDA device.
-        compute_device = torch.device("cuda")
-    return compute_device
+    try:
+        compute_device = next(module.parameters()).device
+        if compute_device.type == "cuda":
+            return compute_device
+    except StopIteration:
+        pass
+    # Fall back to current CUDA device
+    return torch.device("cuda")
```

Comment:
when if compute_device is cpu on line 1544, the old code will return cpu, but now we return cuda? Is that the intent?

Reply:
I think in the old code it should also return "cuda", since in that case (`compute.device.type == "cpu"`) we'd go to line 1546

Reply:
you are right, the old code is `!=`. I misread it.